packages/features/ee/common/server/LicenseKeyService.test.ts

```diff
@@ -0,0 +1,167 @@
+vi.mock("./private-api-utils", () => ({
+  generateNonce: vi.fn(),
+  createSignature: vi.fn(),
+}));
+
+const BASE_HEADERS = {
+  "Content-Type": "application/json",
+};
+
+describe("LicenseKeyService", () => {
+  let service: LicenseKeyService;
+  let baseUrl: string;
+  let licenseKey: string;
+
+  beforeEach(async () => {
+    baseUrl = "http://test-api.com";
+    licenseKey = "test-license-key";
+    process.env.CALCOM_PRIVATE_API_ROUTE = baseUrl;
+    vi.mocked(getDeploymentKey).mockResolvedValue(licenseKey);
+    service = await LicenseKeyService.create();
+  });
+
+  afterEach(() => {
+    vi.resetAllMocks();
+    delete process.env.CALCOM_PRIVATE_API_ROUTE;
+    delete process.env.NEXT_PUBLIC_IS_E2E;
+  });
+
+  describe("create", () => {
+    it("should throw an error if CALCOM_PRIVATE_API_ROUTE is not set and not self-hosted", async () => {
+      delete process.env.CALCOM_PRIVATE_API_ROUTE;
+      await expect(LicenseKeyService.create()).rejects.toThrow("CALCOM_PRIVATE_API_ROUTE is not set");
+    });
+
+    it("should create an instance of LicenseKeyService", async () => {
+      expect(service).toBeInstanceOf(LicenseKeyService);
+    });
+  });
+
+  describe("incrementUsage", () => {
+    it("should call the incrementUsage API and return the response", async () => {
+      const mockResponse = { success: true };
+      const fetchSpy = vi.spyOn(global, "fetch").mockResolvedValue({
+        json: vi.fn().mockResolvedValue(mockResponse),
+      } as any);
+
+      vi.mocked(generateNonce).mockReturnValue("mocked-nonce");
+      vi.mocked(createSignature).mockReturnValue("mocked-signature");
+
+      const response = await service.incrementUsage();
+      expect(response).toEqual(mockResponse);
+      expect(fetchSpy).toHaveBeenCalledWith(`${baseUrl}/v1/license/usage/increment`, {
+        headers: {
+          ...BASE_HEADERS,
+          nonce: "mocked-nonce",
+          signature: "mocked-signature",
+          "x-cal-license-key": "test-license-key",
+        },
+        method: "POST",
+        mode: "cors",
+      });
+    });
+
+    it("should throw an error if the API call fails", async () => {
+      const fetchSpy = vi.spyOn(global, "fetch").mockRejectedValue(new Error("API Failure"));
+
+      vi.mocked(generateNonce).mockReturnValue("mocked-nonce");
+      vi.mocked(createSignature).mockReturnValue("mocked-signature");
+
+      await expect(service.incrementUsage()).rejects.toThrow("API Failure");
+      expect(fetchSpy).toHaveBeenCalledWith(`${baseUrl}/v1/license/usage/increment`, {
+        body: undefined,
+        headers: {
+          ...BASE_HEADERS,
+          nonce: "mocked-nonce",
+          signature: "mocked-signature",
+          "x-cal-license-key": "test-license-key",
+        },
+        method: "POST",
+        mode: "cors",
+      });
+    });
+  });
+
+  describe("checkLicense", () => {
+    it("should return true if NEXT_PUBLIC_IS_E2E is set", async () => {
+      process.env.NEXT_PUBLIC_IS_E2E = "true";
```

Comment:
1. Overriding here won't get consistent results if we're mocking calcom/constants.
2. We use value "1" instead of "true"
3. I think we should override the actual constants per test instead of messing with the global process. This will hurt parallelization.